### PR TITLE
Check if gateway security group in tagged nodes

### LINKED ISSUE #261: OSP: Subctl cloud prepare not happening sometimes
When running "subctl cloud prepare osp ..." on a fresh cluster, it properly configures the gateway node, creates the necessary Security Group rules and associates the SG to the nodes. However, if one of the existing nodes is already labelled as a gateway node and we run "subctl cloud prepare osp ..." it assumes that necessary steps are already done and is not associating the required Security Group rules on the nodes. 

Workaround:

- With `subctl` driven deployment:  delete the `submariner.io/gateway=true` label and re-run the `subctl cloud-prepare` command. 
- With ACM: uninstall the `submariner-addon` from the ManagedCluster and re-install it.

### PR DESCRIPTION
**Pull request by aswinsuryan**
Fixes: #261

Signed-off-by: Aswin Suryanarayanan <aswinsuryan@gmail.com>

### CODE REVIEW

File: pkg/rhos/ocpgwdeployer.go [hunk 1]
```diff
@@ -157,14 +157,22 @@ func (d *ocpGatewayDeployer) Deploy(input api.GatewayDeployInput, status reporte
...
 	gwNodes, err := d.K8sClient.ListGatewayNodes()
 	if err != nil {
 		return status.Error(err, "listing the existing gateway nodes failed")
 	}
 
+	gwNodesList := gwNodes.Items
+	for i := range gwNodesList {
+		err := d.openGatewayPort(groupName, gwNodesList[i].Name, computeClient)
+		if err != nil {
+			return status.Error(err, "failed to open the g/w pods in the existing g/w node")
```
Comment (skitt) [inline pkg/rhos/ocpgwdeployer.go]:
```suggestion
			return status.Error(err, "failed to open the g/w port in the existing g/w node")
```

and add the failing entry to the error message if appropriate.

File: pkg/rhos/ocpgwdeployer.go [hunk 2]
```diff
@@ -157,14 +157,22 @@ func (d *ocpGatewayDeployer) Deploy(input api.GatewayDeployInput, status reporte
...
+		}
+	}
+
+	status.Success("Opened External ports %q in security group %q on RHOS for existing g/w nodes",
```
Comment (skitt) [inline pkg/rhos/ocpgwdeployer.go]:
```suggestion
	status.Success("Opened external ports %q in security group %q on RHOS for existing g/w nodes",
```

**Review by skitt — DISMISSED** [review]

File: pkg/rhos/ocpgwdeployer.go (hunk 2 above)
Comment (tpantelis) [inline pkg/rhos/ocpgwdeployer.go]:
```suggestion
	status.Success("Opened external ports %q in security group %q on RHOS for existing gateway nodes",
```

File: pkg/rhos/ocpgwdeployer.go (hunk 1 above)
Comment (tpantelis) [inline pkg/rhos/ocpgwdeployer.go]:
```suggestion
			return status.Error(err, "failed to open the gateway port in the existing gateway node")
```

**Review by tpantelis — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Pull request merged**